js-pkg/react/src/main.tsx

```diff
@@ -253,26 +259,40 @@ export function YDocProvider(props: YDocProviderProps) {
     let canceled = false
     let provider: YSweetProvider | null = null
     const doc = new Y.Doc()
-
     ;(async () => {
       provider = await createYjsProvider(doc, docId, authEndpoint, {
         initialClientToken,
         offlineSupport: props.offlineSupport,
       })
 
-      if ((props.showDebuggerLink ?? true) && provider.clientToken) {
-        const url = debuggerUrl(provider.clientToken)
-        console.log(
-          `%cOpen this in Y-Sweet Debugger ⮕ ${url}`,
-          'font-size: 1.5em; display: block; padding: 10px;',
-        )
-        console.log(
-          '%cTo hide the debugger link, pass showDebuggerLink={false} to YDocProvider',
-          'font-style: italic;',
-        )
+      // If the client reconnects, the "connected" event will fire again. We should only show the
+      // debugger link again if the client token has changed.
+      let lastClientToken: { value: ClientToken | null } = { value: null }
+
+      let connectionCallback = (status: YSweetStatus) => {
+        if (props.showDebuggerLink === false) return // note: if showDebuggerLink is undefined, we show the debugger link.
```

Comment:
There is potential for a bug here as this closes over the old `props` value. If that `props.showDebuggerLink` value changes, this closure will still use the old value.

Here's an idea: do we only want to show the debugger link for React users? Maybe we should handle this inside the Provider (in `@y-sweet/client`) instead of in React. I think it might be easier to reason about the state there.

Reply:
+1 to showing the link in `@y-sweet/client`.

Not strictly related to this PR but imo one thing that makes this a little confusing is the IIFE (which isn't necessary since `createYjsProvider` isn't an async function)

Reply:
Oh yeah! Totally forgot that we made `createYjsProvider` synchronous!